packages/hebao_v1/contracts/lib/AddressSet.sol

```diff
@@ -23,30 +23,23 @@ contract AddressSet
 {
     struct Set
     {
-        address[] addresses;
+        mapping (uint => address) addresses;
         mapping (address => uint) positions;
         uint count;
```

Comment:
Does it make more sense to leave `addresses` as an array and remove `count`? Seems like `addresses` is used as an array.

Reply:
It will save about 5K gas for each add operation

Reply:
I think it saves 5K gas for each push/pop because the array length doesn't need to be updated, but then you need to update `count` manually which also costs 5K gas. So I think both methods will use the exact same amount of gas because they essentially do the same things (add element to map/array + update `length`/`count`), but with an array it's just a bit cleaner I think. But if you think the current method using a map uses less gas than great, I did not do any measurements.